js/utils/helpers.js

```diff
@@ -1252,32 +1252,61 @@
           return new module.IntegrityConflictError(errorStatusText, mappedErrMessage, generatedErrMessage);
-      }
-      else if (generatedErrMessage.indexOf("violates unique constraint") > -1){
-          var regExp = /\(([^)]+)\)/,
-              matches = regExp.exec(generatedErrMessage), msgTail;
+      } else if (generatedErrMessage.indexOf("violates unique constraint") > -1) {
+          var msgTail, primaryColumns, conflictValues;
+
+          var columnRegExp = /\(([^)]+)\)/,
+              valueRegExp = /=\(([^)]+)\)/,
+              matches = columnRegExp.exec(generatedErrMessage),
+              values = valueRegExp.exec(generatedErrMessage);
 
           if (matches && matches.length > 1) {
-              var primaryColumns =  matches[1].split(','),
-                  numberOfKeys = primaryColumns.length;
+              primaryColumns =  matches[1].split(',');
+              var numberOfKeys = primaryColumns.length;
 
               if (numberOfKeys > 1){
-                msgTail = " combination of " + primaryColumns;
+                  // trim first because sort will put strings with whitespace in front of them before other strings, i.e. " id"
+                  primaryColumns.forEach(function (col, index) {
+                      primaryColumns[index] = col.trim();
+                  });
+
+                  var columnString = "";
+                  // sort the keys because ermrest returns them in a random order every time
+                  primaryColumns.sort();
+                  primaryColumns.forEach(function (col, index) {
+                      columnString += (index != 0 ? ', ' : "") + col;
+                  });
+
+                  msgTail = "combination of " + columnString;
               } else {
-                msgTail = primaryColumns;
+                  msgTail = primaryColumns;
               }
           }
 
+          if (values && values.length > 1) {
+              conflictValues = values[1].split(',');
+          }
 
           mappedErrMessage = "The entry cannot be created/updated. ";
           if (msgTail) {
               mappedErrMessage += "Please use a different "+ msgTail +" for this record.";
           } else {
               mappedErrMessage += "Input data violates unique constraint.";
           }
-          return new module.DuplicateConflictError(errorStatusText, mappedErrMessage, generatedErrMessage);
-      }
-      else{
+
+          // parse out column names and values from generatedErrMessage
+          var conflictKeyValues = [];
+          primaryColumns.forEach(function (colName, idx) {
+              // strip off " character, if present
+              if (colName.indexOf('"') != -1) colName = colName.split('"')[1];
+              conflictKeyValues.push(colName.trim() + "=" + conflictValues[idx].trim());
```

Comment:
I think you need to encode the value and column names. To make sure, you can just change your existing test case to use `duplicate id` instead of `duplicate_id` and change its column type so you can test the encoding too. You don't need to add new test case for it and just changing the existing test should be sufficient.